### PR TITLE
Add Travis Windows build

### LINKED ISSUE #139: Switch from appveyor to travis for windows testing
In my experience, appveyor tends to be flakier and slower than travis. It is currently event disabled for bors. We should try to use travis for windows testing!

### PR DESCRIPTION
**Pull request by phansch**
This adds the Windows build to Travis.

I had to use the `dos2unix` utility on the parser directory to fix some incorrect
line endings. I'm not sure where they are coming from but I guess git is
converting them automatically when cloning on Windows.

Closes #139

### CODE REVIEW

**Comment by phansch** [comment]
I should mention that the windows build takes about 14 minutes because installing `rust-docs` via rustup is really slow on windows.

**Comment by matklad** [comment]
>because installing rust-docs via rustup is really slow on windows.

Can we somehow tell rustup not to download docs?

**Comment by phansch** [comment]
It's not possible, yet. I can't find the most recent discussion right now, but it's being tracked in rustup: https://github.com/rust-lang-nursery/rustup.rs/issues/906.

**Comment by matklad** [comment]
Hm, I think a good idea then is to run windows build only on the master branch, and not gate CI on windows build.

And I think we might remove appveyour.yml and the badge from readme?

**Comment by phansch** [comment]
Should be ready now. I moved the windows build to `allow_failiure`, which means master will be green once the rust linux and npm tests have passed.

Not sure why it is still showing the status for AppVeyor on cb659cd, maybe it has to be merged first.

**Comment by phansch** [comment]
Ok, looks like AppVeyor picked up on the missing appveyor.yml and fails accordingly :laughing: 

```
The build phase is set to "MSBuild" mode (default), but no Visual Studio project or solution files were found in the root directory.
```

File: .travis.yml [hunk 1]
```diff
@@ -29,6 +33,12 @@ matrix:
         - cd editors/code && npm ci && npm run travis
 
   allow_failures:
+    - os: windows
+      if: branch = master
```
Comment (matklad) [inline .travis.yml]: Hm, will it actually include this in the test matrix? Let's see I guess :)

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
Hm, it doesn't actually build windows on master:

image

Perhaps we need to add something like

```
 - os: windows
 - branch: master
```

to the matrix?

**Comment by CAD97** [comment]
Yep, you'll need to keep the build job in the `matrix.include`. The same formulation as in the `allow_failures` should work:

```yml
- os: windows
  if: branch = master
```

Then I'd change `allow_failures` to just be

```yml
- os: windows
```

**Pull request merged**